### ISSUE TITLE
Enumeration-Integer comparison

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#860]
**Reported by sjoelund&#x2e;se on 8 Oct 2012 09:15 UTC**
Modelica.Electrical.Analog.Examples.AD_DA_conversion uses `trig == 4`, but trig is of enumeration type.

I will try to automate a test to find more of these errors.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/860

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#x2e;se on 8 Oct 2012 11:55 UTC**
ModelicaTest.FluidDissipation.Verifications.PressureLoss.StraightPipe.dp_overall_DPMFLOW:
```
[Modelica 3.2.1/Fluid/Dissipation.mo:5222:9-5225:54:writable]
Error: Cannot resolve type of expression IN_con.roughness == 1. The operands have types enumeration(Neglected, Considered), Integer.
```

I am assuming this is Roughness.Neglected that is intended. And Logic.'1' in the AD_DA_conversion.

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#x2e;se on 8 Oct 2012 12:21 UTC**
Fixed in 0282596405137583b97706b91a2fa2754a29f91e
